#1 User should be able to login

emmygozi opened the pull request:
#### What does this PR do?

#### Description of Task to be completed?

A user should be able to login to acess their account
#### How should this be manually tested?

- clone repository and run `npm install`
- Make a post request to localhost:3000/api/v1/login 
#### What are the relevant pivotal tracker stories?

#### Any background context you want to add?

#### Screenshots

Commit:
feature(user): login
user should be able to login
[start #1]

adex001 commented on server/controller/auth.js:
```diff
@@ -36,5 +38,27 @@ class AuthController {
+  static login(req, res) {
+    const { email, password } = req.body;
+
+    return models.Users.findOne( { where: { email } }).then(async (tryLogUser) => {
```
I suggest you use async await here.. it would have made the code cleaner..

adex001 commented on package.json:
```diff
@@ -29,6 +29,7 @@
     "@babel/preset-env": "^7.4.4",
     "@babel/register": "^7.4.4",
     "bcrypt": "^3.0.6",
```
you can uninstall bcrypt package since we have bcryptjs and make necessary updates.

Commit:
refactor(user): login

refactored login function to async and await from using promises
added try catch block to manage any unforseen errors on login
added debugging tool to better view deb messages
[Fixes #165940973]

Commit:
bug(user): login

added debugging tool to better view debbuging messages
[Fixes #165940973]
